src/state/store.js

```diff
@@ -1,25 +1,91 @@
-import { derived } from 'svelte/store';
+import { derived, get } from 'svelte/store';
 
 import { createStore } from '../utils/create-store';
 
 // FIXME: take care of this dependency cycle.
 import { probeSet } from './telemetry-search'; // eslint-disable-line import/no-cycle
 
-import CONFIG from '../config/firefox-desktop';
+import sharedConfig from '../config/shared';
+import productConfig from '../config/products';
 
-// TODO: move this to the new config.js when 'product' is added.
-const DEFAULT_PROBE_PROCESS = 'content';
+export function getFromQueryString(fieldKey, isMulti = false) {
+  const params = new URLSearchParams(window.location.search);
+  const value = params.get(fieldKey);
+  if (isMulti) {
+    return JSON.parse(value);
+  }
+  return value;
+}
+
+function getDefaultState(
+  { basedOnQueryParams } = { basedOnQueryParams: false }
+) {
+  const state = {};
+
+  state.auth = {
+    isAuthenticated: false,
+    token: undefined,
+  };
+
+  // FIXME: applicationStatus or dashboardMode, not both.
+  state.applicationStatus = 'INITIALIZING';
+  state.dashboardMode = {};
+
+  state.product = window.location.pathname.split('/')[1] || 'firefox';
+  state.probeName = '';
+  state.recordedInProcesses = []; // Provided by the API. List of processes this probe was recording in.
+  state.reference = '';
+  state.route = {};
+
+  // Shared config
+  Object.entries(sharedConfig).forEach(([key, { isMulti, defaultValue }]) => {
+    if (basedOnQueryParams) {
+      state[key] = getFromQueryString(key, isMulti) || defaultValue;
+    } else {
+      state[key] = defaultValue;
+    }
+  });
+
+  // Product config
+  state.productDimensions = {};
+  Object.entries(productConfig[state.product].dimensions).forEach(
+    ([key, { defaultValue }]) => {
+      if (basedOnQueryParams) {
+        state.productDimensions[key] = getFromQueryString(key) || defaultValue;
+      } else {
+        state.productDimensions[key] = defaultValue;
+      }
+    }
+  );
+
+  return state;
+}
+
+export const store = createStore(getDefaultState());
```

Comment:
should this have `basedOnQueryParams: true`?

Reply:
Yes, important catch. Thank you!